common/include/pcl/point_types.h

```diff
@@ -790,12 +790,33 @@ namespace pcl
                                                             has_field<PointT, boost::mpl::_2> > >::type
     { };
 
+    /** \brief Traits defined for ease of use with fields
+     * 
+     * has_<fields to be detected>: struct with `value` datamember defined at compiletime
+     * has_<fields to be detected>_v: constexpr boolean
+     * Has<Fields to be detected>: concept modelling name alias for `enable_if`
```

Comment:
I'm not sure if it is a good idea to use virtually the same names (up to case) for different things. And I also (mildly) dislike using CamelCase for meta-programming stuff. What about something like `enable_if_has_xyz`? Too verbose?

Reply:
The case and naming is what I've seen being used for modelling concepts. `is_<name>` becomes `<Name>`, `has_<name>` becomes `Has<Name>`, etc. We can use something `is_<name>` and it'll fit.

`enable_if...` has the issue that in next iteration, names will need to be changed to adhere to C++ concepts. This way, the concept and type traits are tightly coupled.

```c++
// c++ with concepts
template <HasXY PointT>
returnType some_function();

// or with constraints

template <typename PointT> requires HasXY<PointT>
returnType some_function();
```

Reply:
OK, thanks for the explanation. I need to read up about concepts, haven't looked into the upcoming syntax yet.

> The case and naming is what I've seen being used for modelling concepts.

If you have a link to some example, I'd appreciate.

Reply:
> If you have a link to some example, I'd appreciate.

I found a public [blog post](https://www.fluentcpp.com/2019/08/23/how-to-make-sfinae-pretty-and-robust/) comparing different template ways where you can see the syntax comparison